tools/vms-generator/utils/utils.go

```diff
@@ -1048,6 +1049,19 @@ func GetVMIWithHookSidecar() *v1.VirtualMachineInstance {
 	return vmi
 }
 
+func GetVMIWithVirtxmlHookSidecar() *v1.VirtualMachineInstance {
+	vmi := getBaseVMI(VmiWithVirtxmlHookSidecar)
+	vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse("1024M")
+
+	initFedora(&vmi.Spec)
+	addNoCloudDiskWitUserData(&vmi.Spec, generateCloudConfigString(cloudConfigUserPassword))
+
+	vmi.ObjectMeta.Annotations = map[string]string{
+		"hooks.kubevirt.io/hookSidecars": fmt.Sprintf("[{\"args\": [\"--version\", \"v1alpha2\", \"--args\", \"--disk=iotune.total_bytes_sec=52428800,iotune.total_iops_sec=1000|--edit=all\"], \"image\": \"%s/virt-xml-hook:%s\"}]", DockerPrefix, DockerTag),
+	}
+	return vmi
+}
```

Comment:
Does this actually test the feature, or is it just provided as an utility method for test authors?

Reply:
this generates only the example. I didn't add the testing because they are unit tests and I need to add virt-xml binary in the gocli container. I can do it if necessary. I have already implemented some tests [here](https://github.com/alicefr/kubevirt-hook-virt-xml/blob/main/kubevirt_hook_virt_xml_test.go)